### PR TITLE
Ps/pm 5537/move biometric unlock to state providers

### PR DESCRIPTION
**Pull request by MGibson1**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Migrates whether biometric unlock is on for a given user to state providers. This is done for all clients, so it's a little broader than is typical in impact.

I abandoned previous PR (#7833) due to missing commit issue probably caused by github incident.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: libs/common/src/state-migrations/migrations/26-move-biometric-unlock-to-state-providers.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      }),
+    );
+  }
+
+  async rollback(helper: MigrationHelper): Promise<void> {
+    async function rollbackUser(userId: string, account: ExpectedAccountType) {
+      const biometricUnlock = await helper.getFromUser<boolean>(userId, BIOMETRIC_UNLOCK_ENABLED);
+
+      if (biometricUnlock != null) {
+        account ??= {};
```
Comment (coroiu) [inline libs/common/src/state-migrations/migrations/26-move-biometric-unlock-to-state-providers.ts:45]: :memo: thought: wow I had no idea [Nullish coalescing assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment) was a thing, nice!

**Review by coroiu — COMMENTED** [review]
Mostly nitpicks about naming 👍

File: libs/common/src/platform/biometrics/biometric-state.service.ts [hunk 2]
```diff
@@ -78,18 +93,23 @@ export abstract class BiometricStateService {
...
+  biometricUnlockEnabled$: Observable<boolean>;
   encryptedClientKeyHalf$: Observable<EncString | undefined>;
   requirePasswordOnStart$: Observable<boolean>;
   dismissedRequirePasswordOnStartCallout$: Observable<boolean>;
   promptCancelled$: Observable<boolean>;
   promptAutomatically$: Observable<boolean>;
 
   constructor(private stateProvider: StateProvider) {
+    this.biometricUnlockEnabledState = this.stateProvider.getActive(BIOMETRIC_UNLOCK_ENABLED);
+    this.biometricUnlockEnabled$ = this.biometricUnlockEnabledState.state$.pipe(map(booleanState));
```
[the remark this answers is not in the document]
Comment (MGibson1) [reply libs/common/src/platform/biometrics/biometric-state.service.ts:111]: I didn't know about this! thank you

File: libs/common/src/platform/biometrics/biometric-state.service.spec.ts [hunk 3]
```diff
@@ -154,4 +155,47 @@ describe("BiometricStateService", () => {
...
+  });
+
+  describe("getBiometricUnlockEnabled", () => {
+    it("should return the biometricUnlockEnabled state for the given user", async () => {
+      stateProvider.singleUser.getFake(userId, BIOMETRIC_UNLOCK_ENABLED).nextState(true);
+
+      expect(await sut.getBiometricUnlockEnabled(userId)).toBe(true);
+    });
+
+    it("should return false if the state is not set", async () => {
```
[the remark this answers is not in the document]
Comment (coroiu) [reply libs/common/src/platform/biometrics/biometric-state.service.spec.ts]: These turned out great!

**Review by coroiu — DISMISSED** [review]

File: libs/common/src/state-migrations/migrations/27-move-biometric-unlock-to-state-providers.spec.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  let helper: MockProxy<MigrationHelper>;
+  let sut: MoveBiometricUnlockToStateProviders;
+
+  describe("migrate", () => {
+    beforeEach(() => {
+      helper = mockMigrationHelper(exampleJSON(), 26);
+      sut = new MoveBiometricUnlockToStateProviders(26, 27);
+    });
+
+    it("removes biometricUnlock, dismissedBiometricRequirePasswordOnStartCallout, and biometricEncryptionClientKeyHalf from all accounts", async () => {
```
Comment (JaredSnider-Bitwarden) [inline libs/common/src/state-migrations/migrations/27-move-biometric-unlock-to-state-providers.spec.ts:59]: ⚠️ , This migration is only testing the removal of `biometricUnlock` from accounts - not `dismissedBiometricRequirePasswordOnStartCallout` or `biometricEncryptionClientKeyHalf`

File: libs/common/src/state-migrations/migrations/27-move-biometric-unlock-to-state-providers.spec.ts [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+          otherStuff: "otherStuff2",
+        },
+        otherStuff: "otherStuff3",
+      });
+      expect(helper.set).toHaveBeenCalledWith("user-2", {
+        otherStuff: "otherStuff4",
+      });
+    });
+
+    it("sets dismissedBiometricRequirePasswordOnStartCallout value for account that have it", async () => {
```
Comment (JaredSnider-Bitwarden) [inline libs/common/src/state-migrations/migrations/27-move-biometric-unlock-to-state-providers.spec.ts:73]: ⚠️ ,  `dismissedBiometricRequirePasswordOnStartCallout` should be `biometricUnlock`

**Review by JaredSnider-Bitwarden — COMMENTED** [review]
Great work on this! Two tiny naming issues below

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Review by coroiu — APPROVED** [review]

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Pull request merged**